samples/subsys/fs/littlefs/src/main.c

```diff
@@ -73,8 +73,8 @@ void main(void)
 	!(FSTAB_ENTRY_DT_MOUNT_FLAGS(PARTITION_NODE) & FS_MOUNT_FLAG_AUTOMOUNT)
 	rc = fs_mount(mp);
 	if (rc < 0) {
-		printk("FAIL: mount id %u at %s: %d\n",
-		       (unsigned int)mp->storage_dev, mp->mnt_point,
+		printk("FAIL: mount id %llu at %s: %d\n",
+		       (unsigned long long)mp->storage_dev, mp->mnt_point,
```

Comment:
```suggestion
		printk("FAIL: mount id %p at %s: %d\n",
		       mp->storage_dev, mp->mnt_point,
```

Reply:
Everywhere else yes, but here no: the storage_dev for LittleFS, unfortunately, carries identifier of storage partition which is actually unsigned int: https://github.com/zephyrproject-rtos/zephyr/blob/30b4c29aaf5df9ed7d236394218ecd9b2f96ccef/subsys/fs/littlefs_fs.c#L566, https://github.com/zephyrproject-rtos/zephyr/blob/30b4c29aaf5df9ed7d236394218ecd9b2f96ccef/subsys/fs/littlefs_fs.c#L804)

Reply:
In that case:

```suggestion
		printk("FAIL: mount id %" PRIuPTR " at %s: %d\n",
		       (uintptr_t)mp->storage_dev, mp->mnt_point,
```

Reply:
Done

Reply:
@stephanosio Argh. The clang does not want to play nice with this. Will go back to (unsigned long long) as safe option and will mark this as TODO to fix when the clang gets older.

Reply:
```
error: format specifies type 'unsigned int' but the argument has type 'uintptr_t' (aka 'unsigned long') [-Werror,-Wformat]
```

Hold on, something is not right there ... we should not be seeing that warning.

Reply:
This goes back to #16645, where we defined `__INTPTR_TYPE__` and `__UINTPTR_TYPE__` as `long int` and `long unsigned int`, respectively. These are used by the GCC and Clang headers to typedef `intptr_t` and `uintptr_t`.

https://github.com/zephyrproject-rtos/zephyr/blob/feb0e9f766b1328b45666e5e71134c34ae17b742/include/toolchain/zephyr_stdint.h#L47-L50

By default, x86-32 Clang defines `__INTPTR_TYPE__` as `int`, but we override that to `long int` in `zephyr_stdint.h`, so our `intptr_t` becomes `long int` (aka. `long`), and the Clang `inttypes.h` for x86-32 still defines `PRIuPTR` as `%u` and not `%lu`, so we get this warning/error.

This is yet another type system-related bug and needs to be fixed. I will create an issue for this.

Meanwhile, please feel free to go with your previous implementation for now (would be nice to add a `FIXME` comment though).